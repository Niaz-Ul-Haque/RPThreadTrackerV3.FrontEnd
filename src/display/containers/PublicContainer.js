// #region imports
import React from 'react';
import PropTypes from 'prop-types';
import { Container, Row, Col } from 'reactstrap';
import ReduxToastr from 'react-redux-toastr';
import Public from '../views/public/Public';
// #endregion imports

const propTypes = {
	match: PropTypes.shape({})
};
const defaultProps = {
	match: {}
};

const PublicContainer = ({ match }) => (
	<div className="app flex-row align-items-center">
		<ReduxToastr />
		<Container className="public-threads-container">
			<header>
				<h1 className="text-right">Public Threads for Username</h1>
			</header>
			<Row>
				<Col>
					<Public slug={match.params.slug} />
				</Col>
			</Row>
		</Container>
	</div>
);
PublicContainer.propTypes = propTypes;
PublicContainer.defaultProps = defaultProps;
export default PublicContainer;
