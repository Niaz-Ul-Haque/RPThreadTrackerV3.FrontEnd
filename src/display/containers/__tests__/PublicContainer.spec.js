// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import PublicContainer from '../PublicContainer';
// #endregion imports

// #region mocks
jest.mock('../../views/public/Public', () => () => 'Public');
// #endregion mocks

const createTestProps = propOverrides => ({
	match: {
		params: {
			slug: 'my-slug'
		}
	},
	...propOverrides
});

describe('rendering', () => {
	it('should render valid snapshot', () => {
		const props = createTestProps();
		const jsx = (<PublicContainer {...props} />);
		const element = shallow(jsx);
		expect(element).toMatchSnapshot();
	});
});
