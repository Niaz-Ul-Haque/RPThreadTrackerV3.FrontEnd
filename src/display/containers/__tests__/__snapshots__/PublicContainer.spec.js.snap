// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render valid snapshot 1`] = `
<div
  className="app flex-row align-items-center"
>
  <Connect(ReduxToastr) />
  <Container
    className="public-threads-container"
    tag="div"
  >
    <header>
      <h1
        className="text-right"
      >
        Public Threads for Username
      </h1>
    </header>
    <Row
      tag="div"
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Component
          slug="my-slug"
        />
      </Col>
    </Row>
  </Container>
</div>
`;
